### PR TITLE
Add check properties tests

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Related to https://github.com/Project-MONAI/MONAI/issues/7904.

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: ci/verify_bundle.py [hunk 1]
```diff
@@ -65,52 +58,18 @@ def _get_net_id(bundle: str):
...
-def _check_main_section_optional_key(
-    arg_name: str, necessary_key: str, config: dict, main_section: str = "train", sub_section: str = "trainer"
-):
-    # if `arg_name` is in `sub_section`, its value must be `necessary_key`
-    if arg_name in config[main_section][sub_section]:
-        if necessary_key not in config[main_section]:
-            actual_key = str(config[main_section][sub_section][arg_name]).split("#")[-1]
-            raise ValueError(f"'{main_section}' should have '{necessary_key}', got '{actual_key}'.")
-
+def _produce_fake_weights(config_file: str, meta_file: str, bundle: str, device: str):
```
Comment (yiheng-wang-nv) [inline ci/verify_bundle.py:61]: This function is not used in added property checks. I implement it in case we need it for further tests. cc @MingxueG

**Comment by yiheng-wang-nv** [comment]
the error: https://github.com/Project-MONAI/model-zoo/actions/runs/10039379568/job/27743188733?pr=602#step:7:744 shows there is an issue for bundles that using `checkpointloader`. I will try to fix it in a separate PR (can work with enable `supported_apps` together)

File: models/spleen_ct_segmentation/configs/inference.json [hunk 2]
```diff
@@ -100,7 +100,7 @@
         "sw_batch_size": 4,
         "overlap": 0.5
     },
-    "postprocessing": {
+    "postprocess": {
```
Comment (yiheng-wang-nv) [inline models/spleen_ct_segmentation/configs/inference.json:103]: do this change to see if we can get expected error.
Error happens: https://github.com/Project-MONAI/model-zoo/actions/runs/10039656632/job/27744051207?pr=602#step:7:731
cc @MingxueG

File: ci/verify_bundle.py [hunk 3]
```diff
@@ -310,16 +193,51 @@ def verify_torchscript(
...
+                print("workflow_type: ", workflow_type)
+                workflow = create_workflow(
+                    workflow_type=workflow_type,
+                    bundle_root=bundle_path,
+                    config_file=config_file,
+                    logging_file=os.path.join(bundle_path, "configs/logging.conf"),
+                    meta_file=meta_file,
+                    properties_path=properties_path,
+                )
+            check_result = workflow.check_properties()
```
Comment (ericspod) [inline ci/verify_bundle.py:226]: If "supported_apps" is in `metadata` then `workflow` won't be created.

Comment (yiheng-wang-nv) [reply ci/verify_bundle.py:226]: Thanks @ericspod , I changed here

File: models/spleen_ct_segmentation/configs/inference.json [hunk 4]
```diff
@@ -153,7 +153,8 @@
         "load_path": "$@bundle_root + '/models/model.pt'",
         "load_dict": {
             "model": "@network"
-        }
+        },
+        "map_location": "@device"
```
Comment (yiheng-wang-nv) [inline models/spleen_ct_segmentation/configs/inference.json:157]: change here to fix cpu device issue when create bundle workflow. Other bundles that use "checkpointloader" may also have the same issues. I will update them when updating new monai version. cc @KumoLiu

**Pull request closed without merging**